src/dune_action/examples/race.ml

```diff
@@ -0,0 +1,10 @@
+open Dune_action
+
+(* Should we detect that some file appears in both dependencies and targets? *)
```

Comment:
We should make sure some error is produced. (I expect dune currently reports a dependency cycle? I guess that's acceptable, even if not the nicest)

Reply:
Wait, I just realized: we can't detect any problems here if we don't pass targets from dune to the action. We should pass expected targets from dune and fail if the action attempts to write to any of them. (the need for this is relaxed if sandboxing is used, but this should work without sandboxing too)